airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json

```diff
@@ -1,456 +1,466 @@
+        "default": false,
+        "order": 4,
+        "type": "boolean"
+      },
+      "fetch_thumbnail_images": {
+        "title": "Fetch Thumbnail Images from Ad Creative",
+        "description": "Set to active if you want to fetch the thumbnail_url and store the result in thumbnail_data_url for each Ad Creative.",
+        "default": false,
+        "order": 5,
+        "type": "boolean"
+      },
+      "custom_insights": {
+        "title": "Custom Insights",
+        "description": "A list which contains custom ad statistics entries. Each entry must have a name and can contains fields, breakdowns and/or action_breakdowns. Click on 'Add' to fill this field. For more information on configuring custom insights, refer to the <a href='https://docs.airbyte.com/integrations/sources/facebook-marketing'>docs</a>.",
+        "order": 6,
+        "type": "array",
+        "items": {
+          "title": "InsightConfig",
+          "description": "Config for custom insights",
+          "type": "object",
+          "properties": {
+            "name": {
+              "title": "Name",
+              "description": "The name of the custom insight. This will be used as the Airbyte stream name.",
+              "order": 0,
+              "type": "string"
             },
-            "access_token": {
-                "title": "Access Token",
-                "description": "The value of the generated access token. From your App\u2019s Dashboard, click on \"Marketing API\" then \"Tools\". Select permissions <b>ads_management, ads_read, read_insights, business_management</b>. Then click on \"Get token\". See the <a href=\"https://docs.airbyte.com/integrations/sources/facebook-marketing\">docs</a> for more information.",
-                "order": 3,
-                "airbyte_secret": true,
-                "type": "string"
+            "level": {
+              "title": "Level",
+              "description": "The granularity level for data retrieval of the custom insight from the API.",
+              "default": "ad",
+              "enum": [
+                "ad",
+                "adset",
+                "campaign",
+                "account"
+              ],
+              "order": 1,
+              "type": "string"
             },
-            "include_deleted": {
-                "title": "Include Deleted Campaigns, Ads, and AdSets",
-                "description": "Set to active if you want to include data from deleted Campaigns, Ads, and AdSets.",
-                "default": false,
-                "order": 4,
-                "type": "boolean"
+            "fields": {
+              "title": "Fields",
+              "description": "Use the dropdown menu to add the desired fields for your custom insight.",
+              "default": [],
+              "order": 2,
+              "type": "array",
+              "items": {
+                "title": "ValidEnums",
+                "description": "An enumeration.",
```

Comment:
@ChristoGrab I fixed the acceptance test issue ( I believe) but there's a number of fields whose titles and descriptions aren't helpful like this one. Basically anything whose description is `An enumeration` could be better explained. Mind taking another look at those fields?

Reply:
Once you fix this, run `python main.py spec | jq . > integration_tests/spec.json` to update the test suite to expect the updated spec

for more info on why we are updating this file see [this page](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference/#test-spec) (the whole page is helpful to contextualize acceptance testing)

Reply:
I had to do some digging to figure out where this description was coming from, it turns out it was being generated by the Enum class that comes packaged with Python. Worth noting that these titles/descriptions are never displayed to the user in the Airbyte UI, but I've tried to make them more descriptive for anyone looking through the JSON schema :)